specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts

```diff
@@ -138,23 +138,22 @@ export class BusinessRuleManager<SCHEMA extends AnySchema> {
 
   private processCheckFieldResults(
     fieldName: string & keyof SCHEMA['fields'],
-    results: RA<BusinessRuleResult<SCHEMA> | undefined>
+    results: RA<BusinessRuleResult<SCHEMA>>
   ): RA<BusinessRuleResult<SCHEMA>> {
-    return filterArray(
-      results.map((result) => {
-        if (result === undefined) return undefined;
-        if (!specialFields.has(fieldName)) {
-          const field = this.resource.specifyTable.strictGetField(fieldName);
-          setSaveBlockers(
-            this.resource,
-            field,
-            result.isValid ? [] : [result.reason]
-          );
-        }
-        if (result.isValid) result.action?.();
-        return result;
-      })
-    );
+    const field = this.resource.specifyTable.strictGetField(fieldName);
+
+    if (!specialFields.has(fieldName)) {
+      const saveBlockerMessages = filterArray(
```

Comment:
not related to this PR, but though you might find interesting:
I have stopped using the `filterArray` in my projects now, because I discovered a better version of it at Esri:
instead of `filterArray(someArray.map(...))`, we do `someArray.map(...).filter(isDefined)`

where isDefined looks like:
```ts
const isDefined = <T,>(value:T | undefined): value is T => value !== undefined;
```

pros:
- prettier has an easier job formatting your code to look better because .filter(isDefined) call is just chained at the end, rather than having to wrap your entire .map() in a filterArray()
- besides `isDefined`, you can come up with other helpful type assertion utils (i.e isTruthy)

feel free to do a refactor of specify7 code at some point to replace filterArray() with .filter(isDefined)

Reply:
Good refactoring suggesting!
I would agree that would be a very nice improvement. :clap:

Reply:
cc @CarolineDenis @realVinayak